app/Models/Course.php

```diff
@@ -52,4 +52,60 @@ public function scopeOther($query)
     {
         return $query->orderBy('name', config('course.sort_low_to_hight'))->limit(config('course.course_number_home'));
     }
+
+    public function getLearnersAttribute()
+    {
+        return $this->users()->count();
+    }
+
+    public function getLessonsAttribute()
+    {
+        return $this->lessons()->count();
+    }
+
+    public function getTimesAttribute()
+    {
+        return $this->lessons()->sum('time');
+    }
+
+    public function scopeSearch($query, $data)
```

Comment:
giải thích code đoạn code này

Reply:
đoạn code này dùng để truy vấn dữ liệu

Reply:
teacher:
 "select * from `users` inner join `teacher_course` on `users`.`id` = `teacher_course`.`user_id` where `courses`.`id` = `teacher_course`.`course_id` and `user_id` in (?) and `users`.`deleted_at` is null
Lấy tất cả các khóa học có id trùng với id trong bảng teacher_course và có giáo viên có id  sao cho các course không trùng nhau

Reply:
learner:
"select `courses`.*, (select count(*) from `users` inner join `user_course` on `users`.`id` = `user_course`.`user_id` where `courses`.`id` = `user_course`.`course_id` and `users`.`deleted_at` is null) as `users_count` from `courses` where `courses`.`deleted_at` is null order by `users_count`

Thực hiện nối giữa user và course  trả về các bản ghi gồm tất cả các trường trong bảng course và trường có tên là users_count (là đếm số user ứng với từng course_id)
được sắp xếp theo users_count

Reply:
time:
select sum(`lessons`.`time`) from `lessons` where `courses`.`id` = `lessons`.`course_id` and `lessons`.`deleted_at` is null) as `lessons_sum_time` from `courses` where `courses`.`deleted_at` is null order by `created_at` desc, `users_count` asc, `lessons_sum_time` asc

Nối bảng courses với lessons  trả về bản ghi tất cả các trường trong bảng courses và trường `lessons_sum_time  (là tổng time của lessons theo course_id) và được sắp xếp theo lessons_sum_time

Reply:
tag:
"select * from `tags` inner join `course_tag` on `tags`.`id` = `course_tag`.`tag_id` where `courses`.`id` = `course_tag`.`course_id` and `tag_id` in (?) and `tags`.`deleted_at` is null

 Lấy tất cả các khóa học có id trùng với course_id trong bảng course_tag và tag_id = với những tag_id mà người dùng chọn trên selectbox